src/metabase/models/user.clj

```diff
@@ -213,6 +213,13 @@
        (dissoc :password)
        (assoc :ldap_auth true))))
 
+(s/defn update-ldap-user!
```

Comment:
This is a general problem for all SSO - not just LDAP, so perhaps it's worth generalizing it? #15484 (notice the note in Expected Behavior)

Reply:
Good point, this function can be general. I'll update the other SSO providers to match this behavior in a separate PR so that we can close out #15484